### PR TITLE
NIAD-827: add-conditions-for-aws-deployment

### PR DESCRIPTION
**Pull request by RIashfaqahmed**
## Description

Add conditions for AWS deployment for 'main' branch only

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-857

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [ ] Acceptance Criteria met
- [ ] Commit messages are meaningful
- [ ] Manually tested
- [ ] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [ ] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [ ] Add/update any relevant documentation

### CODE REVIEW

**Review by bartek-sarul — APPROVED** [review]

**Comment by andrewwa-kainos** [comment]
Is there any way we can do as much as possible without _actually_ deploying for PR builds? My hope is to try to mitigate the risk of merging to main only to find we've broken something for the deployment.

File: Jenkinsfile [hunk 1]
```diff
@@ -18,6 +18,7 @@ pipeline {
         ENVIRONMENT_ID = "lab-results-build"
         ECR_REPO_DIR = "lab-results"
         DOCKER_IMAGE = "${DOCKER_REGISTRY}/${ECR_REPO_DIR}:${BUILD_TAG}"
+        AWS_DEPLOY_NON_MAIN_BRANCH = false // false: Skip AWS deployment for all branches other than 'main' ; true: Allow AWS Deployment for all branches
```
Comment (n-r-b) [inline Jenkinsfile:21]: This property name doesn't read right for me, just a thought:
```suggestion
        AWS_DEPLOY_ALL_BRANCHES = false // false: Skip AWS deployment for all branches other than 'main' ; true: Allow AWS Deployment for all branches
```

Comment (andrewwa-kainos) [reply Jenkinsfile:21]: Or invert: `AWS_DELPOY_ONLY_MAIN = true`?

Comment (RIashfaqahmed) [reply Jenkinsfile]: Variable name changed to AWS_DEPLOY_ONLY_MAIN with inverted logic

**Review by n-r-b — APPROVED** [review]

**Review by bartek-sarul — APPROVED** [review]

**Pull request merged**